apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
I've re-read links you've provided more carefully, but still think the same as in my last comment :)

So what do you like me to do here?

I can propose one more option:

* I don't do any changes in ```Drafts.add``` - it still will not return Promise as previously.
* In ```ConversationView.updateDraft``` I do like this:
```
....
Drafts.add(...);
return Drafts.store();
```
* I improve ```Drafts.store``` in the following parts:
 * It returns ```Promise``` as in current patch;
 * It returns ```Promise.resolve()``` if Drafts store has not been modified (no add or delete methods, kind of internal ```isEdited```);
 * It returns current "storing promise" if previous ```Drafts.store``` call has not completed yet.

I think there is no need to take care about the case when previous store action has not completed yet, but Drafts has been modified and store called again - I don't see how it's possible for user to do this and NGA will hide ```store``` anyway. We may need this when we start to continuously save draft, but it's separate patch :)

What do you think @julienw ?